src/ec/packages/ec-component-link/link.html.twig

```diff
@@ -68,25 +71,49 @@
 {# Print the result #}
 
 <a href="{{ _link.path }}" class="{{ _css_class }}"{{ _extra_attributes|raw }}>
-  {%- if _icon.name is not empty %}
-    {% if _link.icon_position == 'before' %}
-      {% include '../ec-component-icon/icon.html.twig' with {
-        icon: _icon,
-        extra_classes: 'ecl-link__icon'
-      } only %}
-      {{- '&nbsp;' }}
-    {%- endif -%}
+  {%- if _icons is defined and _link.icon_position == 'before' -%}
+    {%- for icon in _icons -%}
+      {%- if icon.extra_classes is defined and icon.extra_classes is not empty -%}
+        {%- set icon_extra_classes = icon.extra_classes ~ ' ' ~ icon_extra_classes -%}       
+      {%- endif -%}
+      {%- set _icon = _icon|merge(icon) -%}
```

Comment:
Each icon should already have the correct structure, isn't it? So why not using them directly in the include instead of using a temporary variable?
Also, I'm not really sure about it, but won't the second icon inherit all the properties of the first one, as the variable is not reset?

Reply:
I also see that i haven't done this consistently.
The extra_classes have to be extracted, but for the rest it looks like i lost a bit the focus with all this icon _icon _icons :)